application/forms/EventRuleConfigForm.php

```diff
@@ -0,0 +1,647 @@
+        }
+
+        $formValues = [];
+        $formValues['prefixes-map'] = $this->getPrefixesMap(count($values['rule_escalation']));
+
+        foreach ($values['rule_escalation'] as $position => $escalation) {
+            $conditions = explode('|', $escalation['condition'] ?? '');
+            $conditionFormValues = [];
+            $conditionFormValues['condition-count'] = count($conditions);
+            $conditionFormValues['id'] = $escalation['id'] ?? bin2hex(random_bytes(4));
+
+            foreach ($conditions as $key => $condition) {
+                if ($condition === '' && ! isset($formValues['zero-condition-escalation'])) {
+                    $formValues['zero-condition-escalation'] = bin2hex($position);
+                    $conditionFormValues['condition-count'] = 0;
+
+                    continue;
+                }
+
+                $count = $key + 1;
+
+                /** @var Condition $filter */
+                $filter = QueryString::parse($condition);
+                $conditionFormValues['column_' . $count] = $filter->getColumn() === 'placeholder'
+                    ? null
+                    : $filter->getColumn();
+
+                if ($conditionFormValues['column_' . $count]) {
+                    $conditionFormValues['type_' . $count] = $conditionFormValues['column_' . $count];
+                }
+
+                $conditionFormValues['operator_' . $count] = QueryString::getRuleSymbol($filter);
+                $conditionFormValues['val_' . $count] = $filter->getValue();
+            }
+
+            $formValues['escalation-condition_' . bin2hex($position)] = $conditionFormValues;
+            $recipientFormValues = [];
+            if (isset($escalation['recipients'])) {
+                $recipientFormValues['recipient-count'] = count($escalation['recipients']);
+                foreach ($escalation['recipients'] as $key => $recipient) {
+                    if (is_array($recipient)) {
+                        $count = 0;
+                        foreach ($recipient as $elementName => $elementValue) {
+                            if ($elementValue === null) {
+                                continue;
+                            }
+
+                            $count = $key + 1;
+                            $selectedOption = str_replace('id', $elementValue, $elementName, $replaced);
+                            if ($replaced && $elementName !== 'channel_id') {
+                                $recipientFormValues['column_' . $count] = $selectedOption;
+                            } elseif ($elementName === 'channel_id') {
+                                $recipientFormValues['val_' . $count] = $elementValue;
+                            }
+                        }
+
+                        if (isset($recipient['id'])) {
+                            $recipientFormValues['id_' . $count] = (int) $recipient['id'];
+                        }
+                    }
+                }
+            }
```

Comment:
Now I'm sure something is off here. Much of this is split here and in `EscalationCondition` as well as `EscalationRecipient`.

It's bad that processing of input and output is split that way. There should be only one location where this is done, and these are the iframes to me. They accept and provide form data as well as config. They know best what structure is needed where. No need to spread so much *intimate* knowledge around.